scripts/eslint-plugin/rules/no-instanceof-uncached-node.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+module.exports = {
+    meta: {
+        type: 'problem',
+        docs: {
+            description: 'Avoid using uncached Node reference for instanceof test',
```

Comment:
I would generalize this rule into simply "don't use the global `Node`". This catches a few more cases that could potentially throw, such as:

https://github.com/salesforce/lwc/blob/a63caa60bf7863e848ea3883bdfa832c4b257d20/packages/%40lwc/synthetic-shadow/src/env/node.ts#L20

https://github.com/salesforce/lwc/blob/a63caa60bf7863e848ea3883bdfa832c4b257d20/packages/%40lwc/synthetic-shadow/src/faux-shadow/shadow-root.ts#L76

...etc.

(Technically these examples won't throw because they all run immediately when the script loads, but we may introduce breakages in the future this way.)

Plus we might as well do it everywhere and get the (admittedly tiny) minifier savings.